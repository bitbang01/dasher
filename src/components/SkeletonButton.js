import { number, oneOfType, string, func } from 'prop-types'
import styled from 'react-emotion'

import {
  spacing,
  fontSizes,
  fontWeights,
  lineHeights,
  colors,
  radii,
  transition,
  breakpoints,
} from '../theme'
import { joinSpacing, toAlpha } from '../utils/style'

const SkeletonButton = styled.button(props => ({
  width: props.width,
  padding: joinSpacing(spacing[4], 0),
  fontFamily: 'inherit',
  fontWeight: fontWeights.bold,
  fontSize: fontSizes[2],
  lineHeight: lineHeights.normal,
  color: toAlpha(colors.gray[6]),
  backgroundColor: 'transparent',
  border: `2px dashed ${toAlpha(colors.gray[4])}`,
  borderRadius: radii[1],
  cursor: 'pointer',
  outline: 0,
  transition: `background-color ${transition.duration} ${transition.easing}`,

  ':hover,:focus': {
    backgroundColor: toAlpha(colors.gray[1]),
  },

  [breakpoints.sm]: {
    padding: joinSpacing(spacing[6], 0),
  },
}))

SkeletonButton.propTypes = {
  width: oneOfType([number, string]),
  onClick: func,
}

SkeletonButton.defaultProps = {
  width: '100%',
  onClick: () => {},
}

export default SkeletonButton
