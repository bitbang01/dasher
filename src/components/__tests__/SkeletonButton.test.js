import React from 'react'
import { renderIntoDocument, render, fireEvent } from 'react-testing-library'

import SkeletonButton from '../SkeletonButton'

it('renders without crashing', () => {
  const { container } = render(
    <SkeletonButton>Test SkeletonButton</SkeletonButton>,
  )
  expect(container.firstChild).toMatchSnapshot()
})

it('renders with custom number width', () => {
  const { container } = render(
    <SkeletonButton width={360}>Custom Width SkeletonButton</SkeletonButton>,
  )
  expect(container.firstChild).toMatchSnapshot()
})

it('renders with custom percentage width', () => {
  const { container } = render(
    <SkeletonButton width={'100%'}>Custom Width SkeletonButton</SkeletonButton>,
  )
  expect(container.firstChild).toMatchSnapshot()
})

it('calls onClick when button is clicked', () => {
  const handleClick = jest.fn()
  const { getByText } = renderIntoDocument(
    <SkeletonButton onClick={handleClick}>
      Clickable SkeletonButton
    </SkeletonButton>,
  )
  fireEvent(
    getByText('Clickable SkeletonButton'),
    new MouseEvent('click', {
      bubbles: true,
      cancelable: true,
    }),
  )
  expect(handleClick).toHaveBeenCalledTimes(1)
})
