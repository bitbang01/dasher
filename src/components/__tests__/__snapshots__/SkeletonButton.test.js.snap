// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with custom width 1`] = `
.css-0,
[data-css-0] {
  width: 360px;
  padding: 24px 0;
  font-family: inherit;
  font-weight: 600;
  font-size: 16px;
  line-height: 1.5;
  color: rgba(0, 17, 34, 0.47);
  background-color: transparent;
  border: 2px dashed rgba(0, 31, 62, 0.19);
  border-radius: 6px;
  cursor: pointer;
  outline: 0;
  transition: background-color 150ms ease-in-out;
  -webkit-transition: background-color 150ms ease-in-out;
  -moz-transition: background-color 150ms ease-in-out;
}

.css-0:hover,
[data-css-0]:hover,
.css-0:focus,
[data-css-0]:focus {
  background-color: rgba(0, 36, 73, 0.05);
}

@media screen and (min-width: 32em) {
  .css-0,
  [data-css-0] {
    padding: 64px 0;
  }
}

<button
  class="css-0"
>
  Custom Width SkeletonButton
</button>
`;

exports[`renders without crashing 1`] = `
.css-0,
[data-css-0] {
  width: 100%;
  padding: 24px 0;
  font-family: inherit;
  font-weight: 600;
  font-size: 16px;
  line-height: 1.5;
  color: rgba(0, 17, 34, 0.47);
  background-color: transparent;
  border: 2px dashed rgba(0, 31, 62, 0.19);
  border-radius: 6px;
  cursor: pointer;
  outline: 0;
  transition: background-color 150ms ease-in-out;
  -webkit-transition: background-color 150ms ease-in-out;
  -moz-transition: background-color 150ms ease-in-out;
}

.css-0:hover,
[data-css-0]:hover,
.css-0:focus,
[data-css-0]:focus {
  background-color: rgba(0, 36, 73, 0.05);
}

@media screen and (min-width: 32em) {
  .css-0,
  [data-css-0] {
    padding: 64px 0;
  }
}

<button
  class="css-0"
>
  Test SkeletonButton
</button>
`;
